src/main/java/org/folio/rest/impl/DataExportImpl.java

```diff
@@ -122,21 +123,18 @@ public void postDataExportQuickExport(QuickExportRequest entity, Map<String, Str
     getJobProfileForQuickExport(entity)
       .onSuccess(jobProfile ->
         getFileDefinitionForQuickExport(entity, jobProfile.getId(), params, asyncResultHandler)
-          .onSuccess(requestFileDefinition ->
+          .onSuccess(jobData ->
             mappingProfileService.getById(jobProfile.getMappingProfileId(), params)
               .onSuccess(mappingProfile ->
-                jobExecutionService.getById(requestFileDefinition.getJobExecutionId(), tenantId)
-                  .onSuccess(jobExecution ->
-                    jobExecutionService.update(jobExecution.withJobProfileId(jobProfile.getId()), tenantId)
-                      .onSuccess(updatedJobExecution -> {
-                        inputDataManager.init(JsonObject.mapFrom(buildExportRequest(requestFileDefinition.getId(), jobProfile.getId(), entity)), JsonObject.mapFrom(requestFileDefinition), JsonObject.mapFrom(mappingProfile), JsonObject.mapFrom(updatedJobExecution), okapiHeaders);
-                        succeededFuture()
-                          .map(PostDataExportQuickExportResponse.respond200WithApplicationJson(new QuickExportResponse()
-                            .withJobExecutionId(jobExecution.getId())
-                            .withJobExecutionHrId(jobExecution.getHrId())))
-                          .map(Response.class::cast)
-                          .onComplete(asyncResultHandler);
-                      }).onFailure(ar -> failToFetchObjectHelper(ar.getMessage(), asyncResultHandler)))
+                jobExecutionService.update(jobData.getJobExecution().withJobProfileId(jobProfile.getId()), tenantId)
+                  .onSuccess(updatedJobExecution -> {
+                    inputDataManager.init(JsonObject.mapFrom(buildExportRequest(jobData.getFileDefinition().getId(), jobProfile.getId(), entity)), JsonObject.mapFrom(jobData.getFileDefinition()), JsonObject.mapFrom(mappingProfile), JsonObject.mapFrom(updatedJobExecution), okapiHeaders);
```

Comment:
Hi @siarhei-charniak , could you please check usage of jobExecution at init() method as there are cases of retrieving jobExecutions from db after saving like in finalizeExport(). There is could be a case when jobExecution is not found in read node.

Reply:
Good catch, thanks.